server/blob/mail-store/src/test/java/org/apache/james/blob/mail/MimeMessageStoreTest.java

```diff
@@ -0,0 +1,128 @@
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.mail;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.nio.charset.StandardCharsets;
+
+import javax.mail.internet.MimeMessage;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BlobStore;
+import org.apache.james.blob.api.HashBlobId;
+import org.apache.james.blob.api.Store;
+import org.apache.james.blob.memory.MemoryBlobStore;
+import org.apache.james.core.builder.MimeMessageBuilder;
+import org.apache.james.util.MimeMessageUtil;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class MimeMessageStoreTest {
+    private static final HashBlobId.Factory BLOB_ID_FACTORY = new HashBlobId.Factory();
+
+    private Store<MimeMessage, MimeMessagePartsId> testee;
+    private BlobStore blobStore;
+
+    @BeforeEach
+    void setUp() {
+        blobStore = new MemoryBlobStore(BLOB_ID_FACTORY);
+        testee = MimeMessageStore.factory(blobStore).mimeMessageStore();
+    }
+
+    @Test
+    void saveShouldThrowWhenNull() {
+        assertThatThrownBy(() -> testee.save(null))
+            .isInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    void readShouldThrowWhenNull() {
+        assertThatThrownBy(() -> testee.read(null))
+            .isInstanceOf(NullPointerException.class);
+    }
+
+    @Test
+    void mailStoreShouldPreserveContent() throws Exception {
+        MimeMessage message = MimeMessageBuilder.mimeMessageBuilder()
+            .addFrom("any@any.com")
+            .addToRecipient("toddy@any.com")
+            .setSubject("Important Mail")
+            .setText("Important mail content")
+            .build();
+
+        MimeMessagePartsId parts = testee.save(message).join();
+
+        MimeMessage retrievedMessage = testee.read(parts).join();
+
+        assertThat(MimeMessageUtil.asString(retrievedMessage))
+            .isEqualTo(MimeMessageUtil.asString(message));
```

Comment:
Why not compare directly `retrievedMessage` and `message`?

Reply:
Java mail messed up equals and hashCode. Comparing the raw content for MimeMessages is a common practice in the code base